Classes/Middleware/Embedded3dViewer.php

```diff
@@ -210,7 +211,7 @@ public function renderDefaultViewer($model): HtmlResponse
     {
         /** @var ResourceFactory $resourceFactory */
         $resourceFactory = GeneralUtility::makeInstance(ResourceFactory::class);
-        $html = $resourceFactory->retrieveFileOrFolderObject('EXT:dlf/Resources/Private/Templates/Embedded3dViewer/Standalone.html')->getContents();
+        $html = $resourceFactory->retrieveFileOrFolderObject('EXT:dlf/Resources/Public/Html/Embedded3dViewerStandalone.html')->getContents();
```

Comment:
Why did you move this template to `public` folder? It is only handled on the server, so it doesn't need to be accessible by clients. Or am I missing something?

Reply:
This is because changes in TYPO3 12 modify the behavior of handling resources. `ResourceFactory->retrieveFileOrFolderObject` now only handles public files.

> Tried to access a private resource file ... from fallback compatibility storage. This storage only handles public files.

I looked into other TYPO3 extensions, and a public HTML folder seems to be common.

Reply:
OK, thanks for the explanation!

Reply:
For security reasons we can add maybe a `.htaccess` file with deny rules but i think that is not necessary and i didn´t found an example with such a file in public HTML folder.